lib/css_parser/parser.rb

```diff
@@ -684,6 +580,45 @@ def read_remote_file(uri) # :nodoc:
 
   private
 
+    def split_media_query_by_or_condition(media_query_selector)
+      media_query_selector
+        .each_with_object([[]]) do |token, sum|
+          # comma is the same as or
+          # https://developer.mozilla.org/en-US/docs/Web/CSS/@media#logical_operators
+          case token
+          in node: :comma
+            sum << []
+          in node: :ident, value: 'or' # rubocop:disable Lint/DuplicateBranch
+            sum << []
+          else
+            sum.last << token
+          end
+        end # rubocop:disable Style/MultilineBlockChain
+        .map { Crass::Parser.stringify(_1).strip }
+        .reject(&:empty?)
+        .map(&:to_sym)
+    end
+
+    def create_declaration_from_properties(properties)
+      declarations = RuleSet::Declarations.new
+
+      properties.each do |child|
+        case child
+        in node: :property, value: '' # nothing, happen for { color:green; color: }
+        in node: :property
+          declarations.add_declaration!(
+            child[:name],
+            RuleSet::Declarations::Value.new(child[:value], important: child[:important])
+          )
+        in node: :whitespace # nothing
+        in node: :semicolon # nothing
+        in node: :error # nothing
```

Comment:
can we remove these 3 or even better add a "else raise" ?

Reply:
This is pattern patching. If none of the branches matches it will raise `NoMatchingPatternError`. I like that error because adds which node it received

One example
```
NoMatchingPatternError: {:node=>:style_rule, :selector=>{:node=>:selector, :value=>"body", :tokens=>[{:node=>:ident, :pos=>6, :raw=>"body", :value=>"body"}, {:node=>:whitespace, :pos=>10, :raw=>" "}]}, :children=>[{:node=>:whitespace, :pos=>12, :raw=>" "}, {:node=>:property, :name=>"font-size", :value=>"10pt", :children=>[{:node=>:whitespace, :pos=>23, :raw=>" "}, {:node=>:dimension, :pos=>24, :raw=>"10pt", :repr=>"10", :type=>:integer, :unit=>"pt", :value=>10}, {:node=>:whitespace, :pos=>28, :raw=>" "}], :important=>false, :tokens=>[{:node=>:ident, :pos=>13, :raw=>"font-size", :value=>"font-size"}, {:node=>:colon, :pos=>22, :raw=>":"}, {:node=>:whitespace, :pos=>23, :raw=>" "}, {:node=>:dimension, :pos=>24, :raw=>"10pt", :repr=>"10", :type=>:integer, :unit=>"pt", :value=>10}, {:node=>:whitespace, :pos=>28, :raw=>" "}]}]}
```

I think I need to add some more section for stuff like `@keyframe`